### PR TITLE
Serialize UnicodeSetAttributes correctly

### PR DESCRIPTION
**Pull request by jmphilli**
When writing strings to db do not json encode them.

If they were json encoded, decode them with json. Otherwise just return them.

### CODE REVIEW

File: pynamodb/attributes.py [hunk 1]
```diff
@@ -66,44 +66,47 @@ def get_value(self, value):
 class SetMixin(object):
     """
     Adds (de)serialization methods for sets
+
+    def element_serialize(self, value):
+        raise NotImplemented()
+
+    def element_deserialize(self, value):
+        raise NotImplemented()
```
Comment (jmphilli) [inline pynamodb/attributes.py]: @danielhochman how do i do this? how do i 🐍 ?

Comment (jmphilli) [inline pynamodb/attributes.py]: oh i just switch the import order don't i?

**Comment by danielhochman** [comment]
i think this could be cleaner by not requiring redefinition of serialization functions we already have on for the main attributes.

```
class MyMixin(object):
    def serialize_stuff(self):
        self.attr_class.serialize()

class BaseThing(object):
    @staticmethod
    def serialize():
        print 'ok'


class MyThing(MyMixin, BaseThing):
    attr_class = BaseThing

m = MyThing()

m.serialize_stuff()```

it requires the serialization method in the attributes to be static but i think that's an okay change. in this way we can call the original serializer rather than rewriting it.

seems no matter what we are going to break backwards compat if we try and fix this. i would vote we do the above or revert back to the other change where we override the Mixin serialize and leave the Mixin class for b, and can do the backwards compat breaking change in a later version.

**Comment by danielhochman** [comment]
also i think there is a way to do this without breaking backwards compat at all but it would require pulling stuff out of `locals()` which is pretty hacky.

**Comment by danielhochman** [comment]
+1

**Pull request merged**